tests/Unit/Evolution/DgSubcell/Test_Reconstruction.cpp

```diff
@@ -0,0 +1,239 @@
+            Affine(-1.0, 1.0, 2.0, 5.0));
+      },
+      [](std::integral_constant<size_t, 2> /*meta*/) {
+        using domain::make_coordinate_map_base;
+        using domain::CoordinateMaps::Affine;
+        using domain::CoordinateMaps::Wedge2D;
+        using domain::CoordinateMaps::ProductOf2Maps;
+        return make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
+            ProductOf2Maps<Affine, Affine>(Affine(-1.0, 1.0, -1.0, -0.8),
+                                           Affine(-1.0, 1.0, -1.0, -0.8)),
+            Wedge2D(0.5, 0.75, 1.0, 1.0, {}, false));
+      },
+      [](std::integral_constant<size_t, 3> /*meta*/) {
+        using domain::make_coordinate_map_base;
+        using domain::CoordinateMaps::Affine;
+        using domain::CoordinateMaps::Wedge3D;
+        using domain::CoordinateMaps::ProductOf3Maps;
+        return make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
+            ProductOf3Maps<Affine, Affine, Affine>(
+                Affine(-1.0, 1.0, -1.0, -0.8), Affine(-1.0, 1.0, -1.0, -0.8),
+                Affine(-1.0, 1.0, 0.8, 1.0)),
+            Wedge3D(0.5, 0.75, {}, 1.0, 1.0, false));
+      })(std::integral_constant<size_t, Dim>{});
+  const auto& coord_map = *coord_map_base_ptr;
+
+  const size_t start_pt =
+      std::max(static_cast<size_t>(2),
+               Spectral::minimum_number_of_points<BasisType, QuadratureType>);
+  if (eps.size() != 1 and (eps.size() != MaxPts + 1 - start_pt)) {
+    ERROR("Eps is the wrong size. Must be size 1 or size "
+          << MaxPts + 1 - start_pt << " but is " << eps.size());
+  }
+  for (size_t num_pts_1d = start_pt; num_pts_1d < MaxPts + 1; ++num_pts_1d) {
+    const double local_eps =
+        eps.size() == 1 ? eps[0] : eps[num_pts_1d - start_pt];
+    Approx local_approx = Approx::custom().epsilon(local_eps).scale(1.);
+    CAPTURE(local_eps);
+    CAPTURE(num_pts_1d);
+    const Mesh<Dim> dg_mesh{num_pts_1d, BasisType, QuadratureType};
+    const auto inertial_coords = coord_map(logical_coordinates(dg_mesh));
+    const size_t num_subcells_1d = 2 * num_pts_1d - 1;
+    const Mesh<Dim> subcell_mesh(num_subcells_1d,
+                                 Spectral::Basis::FiniteDifference,
+                                 Spectral::Quadrature::CellCentered);
+
+    const auto subcell_logical_coords = logical_coordinates(subcell_mesh);
+    const auto subcell_inertial_coords = coord_map(subcell_logical_coords);
+
+    const auto dg_jac = coord_map.jacobian(logical_coordinates(dg_mesh));
+    const DataVector dg_det_jac = get(determinant(dg_jac));
+    const DataVector projected_det_jac = evolution::dg::subcell::fd::project(
+        dg_det_jac, dg_mesh, subcell_mesh.extents());
+
+    // Our FD reconstruction scheme can integrate polynomials up to degree 6
+    // exactly. However, we want to verify that if we have more than 8 grid
+    // points on the DG grid that we still are able to recover the correct
+    // solution.
```

Comment:
Unclear what this sentence about 8 grid points has to do with the test... clarify?

Reply:
This is trying to say that we want to verify that for more than 6 points and for curved meshes in general `R(P(u))=u`. Low-enough degree polynomials this because trivial because the integration is exact.